lib/filterx/object-json-array.c

```diff
@@ -231,13 +243,29 @@ _unset_index(FilterXList *s, guint64 index)
   return TRUE;
 }
 
+static void
+_make_readonly(FilterXObject *s)
+{
+  FilterXJsonArray *self = (FilterXJsonArray *) s;
+
+  /* racy, but the result will be consistent */
```

Comment:
[optional]

We could check `g_atomic_pointer_get(&self->cached_ro_literal)` again after locking to eliminate this raciness.

Reply:
Will do